eng/CredScanSuppression.json

```diff
@@ -27,6 +27,10 @@
                 "NewPassword1!",
                 "Password*",
                 "123",
+                "http://foo:bar@1.1.1.1:3000",
```

Comment:
Any idea why this is flagged as a secret? I'd like to understand it before blindly adding it here.

Reply:
Cred scan flags it as a "General Password". Not sure if we can extract more info than that.
```
[Error]     sdk/core/Azure.Core/tests/HttpEnvironmentProxyTest.cs(145,1): Error CSCAN-GENERAL0060: Found General Password.: {Searcher}CSCAN-GENERAL0060
{Code}See HttpEnvironmentProxyTest.cs line 145 for the code resulting in match
{Info}Found General Password.
{Suggest}Validate file contains secrets, remove, roll credential, and use approved store. For additional information on secret remediation see https://aka.ms/credscan
```

It seems a bit random, though. There are similar strings in the same file that pass the scan, such as "http://foo@1.1.1.5" and "http://foo@1.1.1.1:3000".

Reply:
The format of the URL being suppressed is `http://username:password@url:port` where the others you listed would be interpreted as `http://username@url` and `http://username@url:port` so there's no password involved.

_([ref](https://www.w3.org/Addressing/URL/url-spec.txt))_

Reply:
Thanks for the context @jsquire. Can we attempt to remove the `:bar` from these urls as I'm assuming our tests don't really need to care about that fake password.

Reply:
Double checked with Pavel and the test requires the passes to be there and to be different. [Here's the code](https://github.com/Azure/azure-sdk-for-net/blob/71e5e6a622de97a2acc4d0233109920ad65e0093/sdk/core/Azure.Core/tests/HttpEnvironmentProxyTest.cs#L131-L155).

He suggested using [inline suppressions](https://microsoft.sharepoint.com/teams/CESecEngineering/CredScan/CredScan%20Wiki/Suppression%20Usage%20Examples.aspx?xsdata=MDN8MDF8fDM4YTQ0YTNiNTRkNzRiMGRhOGQ2ZWE0ZmVhMzJjN2Q0fDcyZjk4OGJmODZmMTQxYWY5MWFiMmQ3Y2QwMTFkYjQ3fDF8MHw2Mzc2NDczOTk5ODMxMTk4MDd8R29vZHxWR1ZoYlhOVFpXTjFjbWwwZVZObGNuWnBZMlY4ZXlKV0lqb2lNQzR3TGpBd01EQWlMQ0pRSWpvaUlpd2lRVTRpT2lJaUxDSlhWQ0k2TVRKOQ%3D%3D&sdata=RjNTRGpRMjlOdnExQzRqSVkzTHVMWDdrUjJUYml3R1hBLzY4L3BkWGRQQT0%3D&ovuser=72f988bf-86f1-41af-91ab-2d7cd011db47%2Ccamaiaor%40microsoft.com&OR=Teams-HL&CT=1629216423677#) if we want to avoid increasing the suppression file. Is this something we can use?

Reply:
I don't think I would go with inline suppressions here as it doesn't have great support.

@kinelski could you check to see if it actually needs the entire url or if it is only complaining about the `bar` part and if it is only that part can you change bar to another existing placeholder in the list instead.

Reply:
Thanks for the suggestion. I tested it and it works. I'll rename `bar` and `bar1` to `pwd1` and `pwd2` in another PR so we keep the suppression file as it is. I still need to take a look at Metrics Advisor and Key Vault secrets, though, but they seem to be straightforward.

Reply:
https://github.com/Azure/azure-sdk-for-net/pull/23364